### DISCUSSION TITLE
restricting and routing service requests by geo boundaries

### DISCUSSION THREAD

**Discussion by pwalsh** [govflow/govflow#15] [Ideas]
We are finding that it is common that service requests can be submitted to a jurisdiction, because the requester thinks it is her jurisdiction, but the request actually belongs to an adjacent jurisdiction. There are city lines that are not really clear to citizens, and of course, the citizen just wants to get something fixed, and should not need to hold a mental model of which jurisdiction is responsible.

The solution to this category of issues is primarily methodological - how the jurisdiction handles requests that are not its concern, what relations (formal or otherwise) exist with neighboring jurisdictions, etc. There are some technical approaches to help route requests to the correct destination in these cases, but their applicability and success is entirely dependent on in internal workflows that are not technical. 

A recent discussion with a city raised the idea that polygons of the city boundaries can be put to use around this category of problem. If the city boundary is known by `govflow`, then, on submission of requests, some "smart" handling can be initiated.

Further, information on the boundaries of directly adjacent jurisdictions, and some additional metadata about those jurisdictions, would provide further context on how such requests could be handled (and such handling would need to be configurable).

Some treatment of this would require at very least the following:

1. Provide a polygon for a jurisdiction
    - and ideally neighboring jurisdictions
2. Provide a tolerance level for out of bounds requests
    - the polygon could be inaccurate, for example  
3. Provide policies for out of bounds requests, which use metadata about neighboring jurisdictions, for example:
    - Do we have a primary phone number/url/email for that jurisdiction's requests? If yes, immediately respond to the request with those details.
    - Do we have an Open311 or govflow endpoint for that jurisdiction’s requests? If yes, route it there, process the callback and present relevant info to the user.

**Comment by idoivri** [comment]
To elaborate on 3, I'd say that this plays into the "interoperability" of the platform with other jurisdictions / agencies, and therefore we should allow:
1 - an option to actually receive out of bounds requests and flag them differently
2 - routing based on types / boundaries to other jurisdictions / agencies (like a 311 that receives requests for water bills and fwds them to the water utility).

**Reply by pwalsh** [reply]
Yes, it seems that re-routing requests to a different jurisdiction is a version of routing requests to different departments, at a technical level. Organizationally, there may be other blockers, but at least we should be able to facilitate this use case to the extent possible with a technical platform.

**Reply by idoivri** [reply]
cool
